sdk/playwrighttesting/microsoft-playwright-testing/src/utils/serviceClient.ts

```diff
@@ -60,12 +60,24 @@ export class ServiceClient {
     );
     if (response.status === 200) {
       return JSON.parse(response.bodyAsText!) as TestRun;
-    } else {
-      throw new Error(`Received status ${response.status} from service from GET TestRun call.`);
-    }
+    }  else if (response.status === 400) {
```

Comment:
We could avoid code duplication here by moving error handling to a separate method, and adding cases within it if required for any special handling based on the API call.

Reply:
sure we can, will do it.